tests/30rooms/06membership.pl

```diff
@@ -205,3 +206,157 @@
          );
       });
    };
+
+test "Can invite existing 3pid",
+   requires => [qw( do_request_json more_users test_http_server_uri_base await_http_request )],
+
+   do => sub {
+      my ( $do_request_json, $more_users, $test_http_server_uri_base, $await_http_request ) = @_;
+
+      my $invitee_email = "marmosets\@monkeyworld.org";
+      my $invitee_mxid = $more_users->[0]->user_id;
+      my $room_id;
+
+      Future->needs_all(
+         stub_is_lookup($invitee_email, $invitee_mxid, $await_http_request),
+
+         make_inviteonly_room($do_request_json)
+         ->on_done( sub {
```

Comment:
This dual on_done + then structure isn't necessary, and is noiser to read. Just use

    ->then( sub {
       ( $room_id ) = @_;